monai/data/image_reader.py

```diff
@@ -148,6 +149,25 @@ def _stack_images(image_list: list, meta_dict: dict):
     return np.stack(image_list, axis=0)
 
 
+def update_json(input_file=None, output_file=None):
+        record_path = "img-label.json"
```

Comment:
Hi @staydelight, thank you for the PR. I have a few concerns:
1. Do we really need to make any changes to read images? It seems that we can already support reading paired data using LoadImaged.
2. I suggest adding a flag to `SaveImage` to allow users to choose whether or not to write a JSON file.
3. Can you clarify the purpose of the record_path? We can directly obtain the label path from the input of SaveImage, and for the image path, we can retrieve it from the metadata of the data (since we have introduced MetaTensor).

Let me know if you need further clarification on any of these points.

Reply:
Hi @staydelight thanks as well but I also have concerns similar to @KumoLiu. In the general case users are not going to want these JSON files being generated so I don't think we should add this to the readers and writers at all. This is very use-case specific so writing some custom transform or doing things in a different way would be the solution. 

One other thing to mention is that it's not thread-safe since multiple parallel transforms may be reading/writing the file at the same time. In this case you also cannot rely on the ordering of LoadImage/SaveImage operations to ensure you match the right input with output.

As mentioned you can access original paths with the metadata present in the MetaTensor objects:

```python
trans = monai.transforms.LoadImaged(keys="image")
d = trans({"image": "/path/to/file.nii.gz"})
print(d["image"].meta["filename_or_obj"])
```

It should be possible to access this value in the postprocessing transform sequence were SaveImage is used since the network output should be a MetaTensor with these values included. You should be able to define a transform after SaveImage which logs these values to a file. This would be the much more modular approach versus adding specific code to the loader/saver classes, so I'd strongly suggest investigating how best to go about that.

Reply:
Hi @ericspod @KumoLiu,Thank you for your advice. According to what you said, since the network output should be a MetaTensor, is it possible to add a `save_log` function to `SaveImage` like this:

```python
self.log_data.append({
    "input": meta_data.get("filename_or_obj", "(unknown)"),
    "output": filename
})


def save_log(self):
    try:
        with open(self.log_json_path, 'r') as f:
            existing_log_data = json.load(f)
    except FileNotFoundError:
        existing_log_data = []

    with open(self.log_json_path, 'w') as f:
        existing_log_data.extend(self.log_data)
        json.dump(existing_log_data, f, indent=4)
```

Reply:
It seems like a workable solution, but it involves repeatedly reading and writing JSON files. I haven't thought of a better way yet. We could add a utility function, but it might not be very effective. @ericspod, do you have any better suggestions for saving a mapping?

BTW, the save path can also be record in the meta data:
https://github.com/Project-MONAI/MONAI/blob/4029c422f1da84cdf5e4ce546dbff33245cbe025/monai/transforms/io/array.py#L507

Reply:
This is a similar sort of change we discussed with the input-output mapping file generation in another PR. I think this would be better implemented in a transform which will appear after SaveImage(d) in your pipeline which will handle saving to file(s). The code here can be put into that and keep SaveImage(d) focused on saving the image data only. You would also have to take into account multiprocessing when doing this and write to separate files for each subprocess, this implementation as-is contains a race condition if multiple processes attempt to save to the same file.

Reply:
Hi @KumoLiu @ericspod, I am writing a test file for `MappingJson`. I tried to create a numpy array to simulate an image and mimic the actual usage where `LoadImage` and `SaveImage` would be used to obtain the contents of meta_data. However, during this process, I needed to import external dependencies. In min_tests.py, any files that require importing external dependencies are skipped. Therefore, I would like to ask if I can add my test file to the skipped list, or should I write a test file that does not require importing external dependencies?

Reply:
I would add your file to the skipped list, otherwise you can use the same [pattern with the nrrd tests](https://github.com/Project-MONAI/MONAI/blob/7a8680e84457cb374859639ab6a078313da85926/tests/test_nrrd_reader.py#L49) to make the test optional.